packages/core/codeframe/src/codeframe.js

```diff
@@ -116,102 +125,166 @@ export default function codeFrame(
       : endLine;
   let endLineString = endLine.toString(10);
 
-  let resultLines = [];
+  // Split input into lines and highlight syntax
   let lines = code.split(NEWLINE);
   let syntaxHighlightedLines = (opts.syntaxHighlighting
     ? highlightSyntax(code, opts.language)
     : code
   )
     .replace(TAB_REPLACE_REGEX, TAB_REPLACEMENT)
     .split(NEWLINE);
-  for (let i = startLine; i < lines.length; i++) {
-    if (i > endLine) break;
 
-    let originalLine: string = lines[i];
-    let syntaxHighlightedLine = syntaxHighlightedLines[i];
+  // Loop over all lines and create codeframe
+  let resultLines = [];
+  for (
+    let currentLineIndex = startLine;
+    currentLineIndex < syntaxHighlightedLines.length;
+    currentLineIndex++
+  ) {
+    if (currentLineIndex > endLine) break;
+    if (currentLineIndex > syntaxHighlightedLines.length - 1) break;
+
+    // Find highlights that need to get rendered on the current line
+    let lineHighlights = highlights
+      .filter(
+        highlight =>
+          highlight.start.line <= currentLineIndex &&
+          highlight.end.line >= currentLineIndex,
+      )
+      .sort(
+        (a, b) =>
+          (a.start.line < currentLineIndex ? 0 : a.start.column) -
+          (b.start.line < currentLineIndex ? 0 : b.start.column),
+      );
 
-    let foundHighlights = highlights.filter(
-      highlight => highlight.start.line <= i && highlight.end.line >= i,
-    );
+    // Check if this line has a full line highlight
+    let isWholeLine =
+      lineHighlights.length &&
+      !!lineHighlights.find(
+        h => h.start.line < currentLineIndex && h.end.line > currentLineIndex,
+      );
+
+    let lineLengthLimit = 1000;
```

Comment:
So for example on CI, a line can be 1000 chars long? Maybe `terminalWidth` should default to the traditional 80 columns?

Reply:
Sure will update it, it was just a fail-safe. But I can make it a proper default